cmake/FindCUDA.cmake

```diff
@@ -799,7 +799,11 @@ if(CUDA_VERSION VERSION_GREATER "7.5")
   find_cuda_helper_libs(nppc)
   find_cuda_helper_libs(nppial)
   find_cuda_helper_libs(nppicc)
-  find_cuda_helper_libs(nppicom)
+  if(CUDA_VERSION VERSION_GREATER_EQUAL "11.0")
```

Comment:
It would be far more great if you do NOT hard code the CUDA version here, but check the contents of `CUDA_nppicom_LIBRAY` afterward.
If `nppicom` is not found, the contents of `CUDA_nppicom_LIBRARY` becomes `CUDA_nppicom_LIBRARY-NOTFOUND`.
Adding to `CUDA_nppi_LIBRARY` can happen after checking the variable, and that would make this script much more robust than hard coding the CUDA number.

Reply:
In fact, this script already hard codes the CUDA version. The failure of find_cuda_helper_libs would cause error messages on the screen, and the user would be confused. Now this script is only used before cmake 3.9, eventually it should be removed. So I don't care too much about the coding quality.

Reply:
1. Fine. This PR will work.
2. About following point, I really appreciate if you could respect what @alalek is pointing.

>Now this script is only used before cmake 3.9

If you look in the latest CMake code [FindCUDA.cmake](https://gitlab.kitware.com/cmake/cmake/-/blob/71f6c7badfdce6279442e124ff0f9b956c926c4a/Modules/FindCUDA.cmake), the way that `nppicom` is treated is same as before.
If `nppicom` doesn't exist, it will lead to build failure.

What @alalek [comment](https://github.com/opencv/opencv/pull/17499#issuecomment-644950107) indicate is upgrading this [line](https://github.com/opencv/opencv/blob/56373428dc5370ad704083935f8354a7e88dfe97/cmake/OpenCVDetectCUDA.cmake#L12) from `3.9.0` to `3.18` (or `3.19`) so until CMake officially upgrades the implementation, we can use OpenCV version to compensate.

> We can do something like [this](https://github.com/opencv/opencv/blob/56373428dc5370ad704083935f8354a7e88dfe97/cmake/OpenCVDetectCUDA.cmake#L12-L16) 

https://github.com/opencv/opencv/blob/56373428dc5370ad704083935f8354a7e88dfe97/cmake/OpenCVDetectCUDA.cmake#L12

(Or, at least I interpreted as so)

So, the file you are editing is much more valuable than what you think
> So I don't care too much about the coding quality.

3. No matter which decision you make, I respect your decision and OpenCV team's decision.  I have no rights to force you to follow my comment.

Reply:
@tomoaki0705 I am sorry, I was too tired at that time. It was a joke, don't take it seriously.  IMO, checking that some library is not found by CUDA_xxx_LIBRARY-NOTFOUND and remove it from the library is error-prone and don't provide much information. For example, if I used CUDA 9.0 and removed nppicom accidentally, then the linking error would not occur at the cmake configuration stage and is postponed lately. So we need the CUDA version to help overcome such difficulties.
But in fact, there is no need to change FindCUDA.cmake in this case, We just add the checking after it.
In the long run, we could consider coping and using FindCUDAToolkit.cmake as it provides fine-grained npp targets and is more robust. But I am not sure whether it could be combined with old cmake versions since OpenCV supports some old platforms now and we need a plan  to raise the miminal supported cmake version.

Reply:
Thanks for the reply @cyyever 
> remove it from the library is error-prone and don't provide much information
I feel that now we are talking.
That point was missing from my point of view.
Automatically making things robust may lead to confuse, and that's exactly what we need to review.

Again, I'm fine with this PR now.  It makes sense.